paparazzi/paparazzi/src/main/java/app/cash/paparazzi/accessibility/A11ySnapshotHandler.kt

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.paparazzi.accessibility
+
+import app.cash.paparazzi.Snapshot
+import app.cash.paparazzi.SnapshotHandler
+import java.awt.AlphaComposite
+import java.awt.BasicStroke
+import java.awt.Color
+import java.awt.Composite
+import java.awt.Graphics2D
+import java.awt.image.BufferedImage
+import kotlin.math.max
+
+/**
+ * A Paparazzi SnapshotHandler that renders the snapshot, with a light coloured overlay,
+ * and adjacent to a legend with matching colours.
+ */
+class A11ySnapshotHandler(
```

Comment:
we should probably avoid modifying the snapshot handler. Ideally, a11y render extension should handle all the logic of fetching metadata and displaying it. Also when building the initial render extension, we moved away from using Graphics2D apis for drawing for 2 reasons.

1. Platform consistency across Mac, Linux, and Windows rendering differently.
2. It caused build issues when depending on Paparazzi plugin where java.awt.* classes couldn't be found.

Reply:
OK, I'll take a look at this.  I looked at https://github.com/cashapp/paparazzi/blob/master/paparazzi/paparazzi/src/main/java/app/cash/paparazzi/accessibility/AccessibilityRenderExtension.kt, which works in the world of Views. 

It seemed to be a nice thing, that instead of mutating the View hierarchy, this was able to work in two passes, 

1) Collecting the a11y nodes
2) decorating the untouched Compose snapshot, with accessibility overlays and text. 

But it would potentially solve/simplify the task of matching, as we would avoid Graphics2D.

My one nagging doubt is related to the lifecycle of `ViewRootForTest.onViewCreatedCallback` and `composeView.viewTreeObserver.addOnPreDrawListener`, I'll see what I can do here once I addressed other feedback.